docs/development-process.md

```diff
@@ -0,0 +1,77 @@
+# Rancher2 Provider
+
+Terraform has recently evolved into a spotlight infrastructure provisioning tool that is maintained by Rancher engineering. As its popularity grows, so will its open source development.
+
+Improvements to note since Jan 2023
+* Rancher2 Provider docs
+* Branching into `master 3.x.x` for Rancher 2.7 clusters and `release/v2 2.x.x` for Rancher 2.6 clusters
+* Scripts to test Terraform RCs locally on unix and windows
+
+### Development Process
+
+When developing the provider,
+
+1. Smoke test the feature or bug fix
+2. Merge PR/s
+2. Add a QA test template with your test steps to the Github issue so QA knows how to test it.
+
+```
+# QA Test Template
+
+Issue: <!-- link the issue or issues this PR resolves here -->
+<!-- If your PR depends on changes from another pr link them here and describe why they are needed on your solution section. -->
+  
+# Problem
+<!-- Describe the root cause of the issue you are resolving. This may include what behavior is observed and why it is not desirable. If this is a new feature describe why we need this feature and how it will be used. -->
+  
+# Solution
+<!-- Describe what you changed to fix the issue. Relate your changes back to the original issue / feature and explain why this addresses the issue. -->
+  
+# Testing
+<!-- Note: Confirm if the repro steps in the GitHub issue are valid, if not, please update the issue with accurate repro steps. -->
+ 
+## Engineering Testing
+
+### Manual Testing
+<!-- Describe what manual testing you did (if no testing was done, explain why). -->
+ 
+### Automated Testing
+<!--If you added/updated unit/integration/validation tests, describe what cases they cover and do not cover. -->
+ 
+## QA Testing Considerations
+<!-- Highlight areas or (additional) cases that QA should test -->
+  
+### Regressions Considerations
+<!-- Dedicated section to specifically call out any areas that with higher chance of regressions caused by this change, include estimation of probability of regressions -->
+```
+
+3. Cut a Terraform RC.
+
+Terraform RCs enable QA to test features and bug fixes _before_ the next Terraform version is released. This process is also how Rancher is tested and released. Terraform is expected to mirror Rancher's functionality and use of third party packages because it uses the Rancher API, so it's ideal and best practice to test the provider in the same way.
+
+To cut an RC, tag the latest commit. Bump the minor version for new features and the patch version for bug fixes only.
+
+For example, if the latest tag on `master` is `v3.0.0` and the commits since that tag contain a new feature,
+
+```
+git tag v3.1.0-rc1
+git push upstream v3.1.0-rc1
+```
+
+If all commits since the latest tag are only bug fixes,
+
+```
+git tag v3.0.1-rc1
+git push upstream v3.0.1-rc1
+```
+
+If the latest tag is already an RC, say `v3.0.1-rc1`,
+
+```
+git tag v3.0.1-rc2
+git push upstream v3.0.1-rc2
+```
+
+5. Move the issue To Test
```

Comment:
Because this step requires you to have access to Ranchers Zube community members won't be able to follow this. I think leaving the part about Rancher QA validating the issue is still relevant though.